### ISSUE TITLE
Documentation: Modify the format of the METplus Components Python Requirements Table

### ISSUE THREAD

**Issue by jprestop** [dtcenter/METplus#2463]
## Describe the Task ##
Modify the format of the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.  At the [METplus Engineering meeting on 20240109](https://docs.google.com/document/d/1SwFkyPvVdhUKvG6U-uui6Gty2fNkw5W4FldlI3sM9Vc/edit) (and at previous meetings) it was noted that this table is difficult to read because users have to scroll over and it is hard to keep track of the information in the row.

A drop down format was suggested in place of the table.  Mark up a few rows of the table (to start) in a feature branch to be in a drop down style format and get feedback from the team to see if this is a better way to proceed.  Please start by only doing a few rows and getting feedback from the team, before proceeding with converting the whole table. 

### Time Estimate ###
1 to 3 days of work.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
Per Tara, Lisa should split her time between 
2702691 | JNT DTC NSF BASE FUN…
2784543 | NOAA R2O METp SW2
2799991 | Met Office METplus S…

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by lisagoodrich** [comment]
@jprestop is this ready for a pull request?

**Comment by jprestop** [comment]
@lisagoodrich We are still waiting for feedback from @DanielAdriaansen, so this is not ready for a pull request.

**Comment by DanielAdriaansen** [comment]
Greetings. Thanks for doing this work. I like the dropdown menus. I would make the following suggestions:
1) In each dropdown, remove `METplus Wrappers` if `METplus Wrappers` does not depend on that package. Cartopy is an example. There are use cases that depend on Cartopy, but running METplus Wrappers does not require Cartopy. Perhaps replace with `Select METplus Use Cases`?
2) I still see the dropdown and table listed here: https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/overview.html#metplus-components-python-requirements. I gather this section will be removed in favor of the Appendix?
3) I still see the table in the appendix, and also gather that will be removed in the final version?

Happy to provide review once the PR is ready. Thanks for asking for my input.

**Comment by jprestop** [comment]
@DanielAdriaansen 

Thanks so much for the feedback!

> In each dropdown, remove METplus Wrappers if METplus Wrappers does not depend on that package. Cartopy is an example. There are use cases that depend on Cartopy, but running METplus Wrappers does not require Cartopy. Perhaps replace with Select METplus Use Cases?

I like the idea of replacing "METplus Wrappers" with "Select METplus Use Cases".  @lisagoodrich could you please make that change?  I believe it applies to all references to "METplus Wrappers" except for "dateutil >= 2.82". Please let me know if you have any questions.

> I still see the dropdown and table listed here: https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/overview.html#metplus-components-python-requirements. I gather this section will be removed in favor of the Appendix?

That is correct, Dan. @lisagoodrich could you please go ahead and remove the table at the location referenced above?  Please let me know if you have any questions.

> I still see the table in the appendix, and also gather that will be removed in the final version?

That is also correct, Dan.  @lisagoodrich could you please go ahead and remove the table beneath the dropdowns in the Appendix?  Please let me know if you have any questions.

@georgemccabe The Python version listed under "[Minimum Requirements](https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/installation.html#software-requirements)" is incorrect and the MET version is soon to be incorrect with the release of 12.0.0.  Would you like to propose a change to that text (for Lisa to make) so that we can remove the versions?  Or would you prefer to leave as-is with an update to the Python version and the MET version in this branch?

**Comment by georgemccabe** [comment]
@jprestop, I think Lisa can go ahead and update the python version. It would be nice to not have to update the MET version for each release. Perhaps we could do something fancy in docs/conf.py? Or just add it to the release instructions. For now, I think we should just change it to 12.0.0 since that is the next release.

**Comment by lisagoodrich** [comment]
@jprestop, I've got a quick question, Do you want this 1.5 section completely removed?  Or just linked to the appendix like I have it here?
https://metplus.readthedocs.io/en/feature_2463_modify_table/Users_Guide/overview.html#metplus-components-python-requirements.

**Comment by jprestop** [comment]
@lisagoodrich Thanks for asking.  Since the Appendix is aptly named "Appendix A METplus Components Python Packages" please just remove section 1.5 entirely.

**Comment by jprestop** [comment]
Closed with https://github.com/dtcenter/METplus/pull/2508
